docs/Toolchains.md

```diff
@@ -34,10 +34,24 @@ The workaround is to perform the npm_install inside a docker container so that i
 
 Follow https://github.com/bazelbuild/rules_nodejs/issues/506 for updates on support for node-gyp cross-compilation.
 
-## node_toolchain
+## Registering a custom toolchain
 
-See <./Core.md#node_toolchain>
+To run a custom toolchain (i.e., to run a node binary not supported by the built-in toolchains), you'll need four things:
 
+1) A rule which can build or load a node binary from your repository
+   (a checked-in binary or a build using a relevant [`rules_foreign_cc` build rule](https://bazelbuild.github.io/rules_foreign_cc/) will do nicely).
+2) A [`node_toolchain` rule](Core.html#node_toolchain) which depends on your binary defined in step 1 as its `target_tool`.
+3) A [`toolchain` rule](https://bazel.build/reference/be/platform#toolchain) that depends on your `node_toolchain` rule defined in step 2 as its `toolchain`
+   and on `@rules_nodejs//nodejs:toolchain_type` as its `toolchain_type`. Make sure to define appropriate platform restrictions as described in the
+   documentation for the `toolchain` rule.
+4) A call to [the `register_toolchains` function](https://bazel.build/rules/lib/globals#register_toolchains) in your `WORKSPACE`
+   that refers to the `toolchain` rule defined in step 3.
+
+Examples of steps 2-4 can be found in the [documentation for `node_toolchain`](Core.html#node_toolchain).
+
+If necessary, you can substitute building the node binary as part of the build with using a locally installed version by skipping step 1 and replacing step 2 with:
+
+2) A `node_toolchain` rule which has the path of the system binary as its `target_tool_path`
```

Comment:
Can you add an example of this. I can't find this documented elsewhere.

Reply:
I agree an example would be nice, and if Cockroach wants to fund it then we could make the time to write one.